tests/remote-cache/Earthfile

```diff
@@ -36,13 +36,13 @@ test1:
     RUN nl=$(cat ./output | grep "execute-test1-run-before-copy" | wc -l) && cat /var/log/buildkitd.log && test "$nl" -eq 1
 
     # Cached - should only have 1 line output and not re-echo.
-    RUN nl=$(cat ./output | grep "execute-test1-run-after-copy" | wc -l) && test "$nl" -eq 1
+    RUN nl=$(cat ./output | grep "execute-test1-run-after-copy" | grep -v "*cached*" | wc -l) && test "$nl" -eq 1
```

Comment:
why did this have to change? Is the contents of previous cached lines now output?

Reply:
Here's the output without `--logstream=true`: 
```
+test1 | REGISTRY=172.17.0.4:5000
+test1 | *cached* --> RUN echo "execute-test1-run-before-copy"
+test1 | [----------] 100% RUN echo "execute-test1-run-before-copy"
+test1 | REGISTRY=172.17.0.4:5000
+test1 | --> COPY ./input ./
+test1 | REGISTRY=172.17.0.4:5000
+test1 | --> RUN echo "execute-test1-run-after-copy"
+test1 | execute-test1-run-after-copy
```

And with it disabled: 

```
+test1 | *cached* --> RUN echo "execute-test1-run-before-copy"
+test1 | REGISTRY=172.17.0.4:5000
+test1 | *cached* --> COPY ./input ./
+test1 | REGISTRY=172.17.0.4:5000
+test1 | *cached* --> RUN echo "execute-test1-run-after-copy"
output | --> exporting outputs
```

It seems to be cached, but the `RUN` is being output twice for some reason.

Reply:
I checked out this branch, and reverted this test to the copy from `main` and created a diff (left side is the non-logstream version of earthly, compliled from `main`, and the right side is earthly compiled from your branch).

![image](https://github.com/earthly/earthly/assets/1806823/d11a1e86-4949-4633-b716-1fae028ec952)

The logstream version of earthly is now printing out two lines:

```
# cat ./output | grep "execute-test1-run-after-copy"
              +test1 | *cached* --> RUN echo "execute-test1-run-after-copy"
              +test1 | [----------] 100% RUN echo "execute-test1-run-after-copy"         <------ this being the new one
```

by adding the `grep -v "*cached*"`, it's being changed to validate 

```
+test1 | [----------] 100% RUN echo "execute-test1-run-after-copy" 
```

exists in the output, which breaks what this test is testing for.

instead, this should be changed to something like `grep -v '\[.*\] [0-9]\+%'` to ignore the progress-related output of the command.

Reply:
Thanks for digging. Would it then make more sense to use `-eq 2` since we expect 2 lines here?

Reply:
Actually, the uncached version expects 2 lines. I'll just go with your suggestion.

Reply:
Changing this to `2`, would then cause this test to fail when logstream is disabled -- this test should ideally pass when running both the legacy and logstream mode.